monkey/monkey_island/cc/services/authentication_service/authentication_facade.py

```diff
@@ -71,6 +79,22 @@ def _get_refresh_token_owner(self, refresh_token: ParsedToken) -> User:
             raise Exception("Invalid refresh token")
         return user
 
+    def generate_otp(self) -> str:
+        """
+        Generates a new OTP
+
+        The generated OTP is saved to the `IOTPRepository`
+        :return: The generated OTP
+        """
+        otp_value = self._generate_otp()
+        expiration_time = time.monotonic() + OTP_EXPIRATION_TIME
+        self._otp_repository.insert_otp(otp_value, expiration_time)
+
+        return otp_value
+
+    def _generate_otp(self) -> str:
+        return secure_generate_random_string(32, string.ascii_letters + string.digits + "._-")
```

Comment:
NIT: Not sure if it's worth breaking out `_generate_otp()`

Reply:
If we are going to keep this, though, then:
```suggestion
        otp_value = self._generate_otp()
        expiration_time = time.monotonic() + OTP_EXPIRATION_TIME
        self._otp_repository.insert_otp(otp_value, expiration_time)

        return otp_value

    def _generate_otp(self) -> OTP:
        return secure_generate_random_string(32, string.ascii_letters + string.digits + "._-")
```